libraries/redcore/api/soap/operation/operation.php

```diff
@@ -39,18 +39,12 @@ public function __construct($webservice, $config = array())
 	/**
 	 * Read list
 	 *
-	 * @param   int     $limitStart         Start position number of the fetched data
-	 * @param   int     $limit              Limit the number of fetched rows. Set 0 for all rows
-	 * @param   string  $filterSearch       Search Data with specific filter
-	 * @param   array   $filters            Search Data with specific filter
-	 * @param   string  $ordering           Ordering
-	 * @param   string  $orderingDirection  Ordering direction ASC or DESC
-	 * @param   string  $language           Language Tag name (ex: en)
+	 * @param   object  $data  $limitStart, $limit, $filterSearch,
+	 *                         $filters, $ordering, $orderingDirection, $language
 	 *
 	 * @return  array
 	 */
-	public function readList(
-		$limitStart = 0, $limit = 20, $filterSearch = null, $filters = array(), $ordering = null, $orderingDirection = null, $language = null)
+	public function readList($data)
```

Comment:
I tried not to.  However when using the incoming wsdl file to initialize the SoapServer, it gave me no option.

Plus yes, I am against the array for exposing it for the end users (fields are specific there), but I was never against it for internal processing and I had no choice here.